message/infrastructure/nats/streaming.go

```diff
@@ -0,0 +1,35 @@
+package nats
+
+import "github.com/nats-io/go-nats-streaming"
+
+type NatsStreamingConfig struct {
+	// ClusterID is the NATS Streaming cluster ID.
+	ClusterID string
+
+	// ClientID is the NATS Streaming client ID to connect with.
+	// ClientID can contain only alphanumeric and `-` or `_` characters.
+	//
+	// Using DurableName causes the NATS Streaming server to track
+	// the last acknowledged message for that ClientID + DurableName.
+	ClientID string
+
+	// StanOptions are custom []stan.Option passed to the connection.
+	// It is also used to provide connection parameters, for example:
+	// 		stan.NatsURL("nats://localhost:4222")
+	StanOptions []stan.Option
+
+}
+
+func NewStanConnection(config *NatsStreamingConfig) (stan.Conn,error) {
```

Comment:
what do you thinkg about getting rid of `NewStanConnectionWithDefaultSubscriberOption` and setting these options in `NewStanConnection`? It will be harder to create "wrong" connection for subscriber with NewStanConnection